### PR TITLE
FM2-632: Medication Request should return Null when Drug Order has dose units but no dose

### PR DESCRIPTION
**Pull request by mherman22**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-632

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java [hunk 1]
```diff
@@ -54,6 +54,9 @@ public Dosage toFhirResource(@Nonnull DrugOrder drugOrder) {
 		if (drugOrder.getDose() != null || drugOrder.getDoseUnits() != null) {
 			Dosage.DosageDoseAndRateComponent doseAndRate = new Dosage.DosageDoseAndRateComponent();
 			Quantity dose = new SimpleQuantity();
+			if (drugOrder.getDose() == null) {
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java:57]: We don't want to return null entirely for the dosage if the dose is null, I think it should be sufficient to just change line 54 to "if (drugOrder.getDose() != null) {" ....

**Review by mogoodrich — CHANGES_REQUESTED** [review]
Thanks @mherman22 , I added a comment...

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java (hunk 1 above)
Comment (mherman22) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java]: updated!

**Review by njiddasalifu — COMMENTED** [review]
from small exp bg, @mherman22 i think the request by @mogoodrich is the ideal. thanks

**Review by mogoodrich — APPROVED** [review]
Thanks @mherman22 , looks good to me, though I'm interested in @ibacher 's thoughts before merging.

**Review by ibacher — APPROVED** [review]

**Comment by denniskigen** [comment]
@ibacher, @mogoodrich, do we need a new release of the FHIR module to get this fix generally available?

**Pull request merged**